fem/coefficient.hpp

```diff
@@ -419,18 +419,20 @@ class GridFunctionCoefficient : public Coefficient
     where T is the transformation rule, and Q1/Q2 are the parent coefficients.*/
 class TransformedCoefficient : public Coefficient
 {
+   using t1_f = std::function<double(double)>;
+   using t2_f = std::function<double(double, double)>;
+
 private:
    Coefficient * Q1;
    Coefficient * Q2;
-   double (*Transform1)(double);
-   double (*Transform2)(double,double);
+   t1_f Transform1;
+   t2_f Transform2;
 
 public:
-   TransformedCoefficient (Coefficient * q,double (*F)(double))
-      : Q1(q), Transform1(F) { Q2 = 0; Transform2 = 0; }
-   TransformedCoefficient (Coefficient * q1,Coefficient * q2,
-                           double (*F)(double,double))
-      : Q1(q1), Q2(q2), Transform2(F) { Transform1 = 0; }
+   TransformedCoefficient (Coefficient * q, t1_f F)
+      : Q1(q), Transform1(std::move(F)) { Q2 = 0; Transform2 = 0; }
```

Comment:
I like the change to `std::function`; it looks good. I wonder if it was safer not to use `std::move` here. I understand that `std::move` may benefit performance, but defining a transformation function is probably only a tiny fraction of the overall application runtime. Would it make sense to use `Transform1(F)` instead of `Transform1(std::move(F))`? I'm mostly suggesting that because I'm unsure what happens to the function once it's moved or, more specifically, if someone uses it after calling this constructor.

Reply:
Since it's passed by value into the constructor, this `std::move` here just allows the compiler to optimize away a second copy (`clang-tidy` typically recommends this syntax). Users constructing a `TransformedCoefficient` aren't required to pass their function in as an rvalue with `std::move` and I don't think this impacts the function in their calling scope.